sdk/core/src/policies/retry_policies/exponential_retry.rs

```diff
@@ -0,0 +1,79 @@
+use crate::policies::{Context, Policy, PolicyResult, Request, Response};
+use crate::sleep::sleep;
+use chrono::{DateTime, Local};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Retry policy with exponential backoff.
+///
+/// Retry policy with exponential backoff (with an added random delay up to 256 ms). Each retry
+/// will happen at least after an exponential wait time. So if x is the first retry wait, the
+/// second will be x*2, the third x*4 and so on. The policy will retry until the maximum number of
+/// retries have been reached or the maximum allowed delay has passed (whichever comes first). The
+/// wait time is not precise.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ExponentialRetryPolicy {
```

Comment:
Re: this and linear retry.

All our SDKs use linear or exponential based on ClientOptions: https://github.com/Azure/azure-sdk-for-net/blob/8d19a820660c3c34b6ebe7904bd4e7ff895c82f6/sdk/core/Azure.Core/src/ClientOptions.cs#L44 and https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/RetryOptions.cs. Thus, how we retry is typically left up to the caller, but the client library can use one or the other, but if we have separate classes we should make them internal and choose the appropriate one based on `RetryOptions`, if having separate classes is easier (see https://github.com/Azure/azure-sdk-for-net/blob/8d19a820660c3c34b6ebe7904bd4e7ff895c82f6/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L181-L200 for how we do this in .NET).